kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/timeout/ContainerTimeoutOverriddenMainDispatcherTest.kt

```diff
@@ -0,0 +1,45 @@
+package com.sksamuel.kotest.engine.test.timeout
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.testCoroutineScheduler
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import kotlinx.coroutines.withContext
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+
+// Issue: https://github.com/kotest/kotest/issues/3703
+@OptIn(ExperimentalCoroutinesApi::class)
+class ContainerTimeoutOverriddenMainDispatcherTest : FunSpec({
```

Comment:
```suggestion
// Issue: https://github.com/kotest/kotest/issues/3703
@DoNotParallelize
@OptIn(ExperimentalCoroutinesApi::class)
class ContainerTimeoutOverriddenMainDispatcherTest : FunSpec({
```

Using/changing `Dispatchers.Main` depends on a global object. This does not play well with any type of concurrency. `@DoNotParallelize` will run this test in isolation, so that it cannot interfere with possible tests elsewhere which also happen to use/change `Dispatchers.Main`.